management/mailconfig.py

```diff
@@ -547,6 +547,47 @@ def get_required_aliases(env):
 
 	return aliases
 
+# multi-factor auth
+
+def get_mfa_state(email, env):
+	c = open_database(env)
+	c.execute('SELECT secret, mru_token FROM totp_credentials WHERE user_email=?', (email,))
+
+	credential_row = c.fetchone()
+	if (credential_row == None):
+		return { 'type': None }
+
+	return {
+		'type': 'totp',
+		'secret': credential_row[0],
+		'mru_token': credential_row[1]
+	}
+
+def create_totp_credential(email, secret, token, env):
+	validate_totp_secret(secret)
+
+	conn, c = open_database(env, with_connection=True)
+	c.execute('INSERT INTO totp_credentials (user_email, secret, mru_token) VALUES (?, ?, ?)', (email, secret, token))
+	conn.commit()
+	return "OK"
+
+def set_mru_totp_code(email, token, env):
+	conn, c = open_database(env, with_connection=True)
+	c.execute('UPDATE totp_credentials SET mru_token=? WHERE user_email=?', (token, email))
+
+	if c.rowcount != 1:
+		conn.close()
+		raise ValueError("That's not a user (%s)." % email)
```

Comment:
According to [PEP 3101](https://www.python.org/dev/peps/pep-3101/) it's better to use .format

```suggestion
		raise ValueError("That's not a user ({}).".format(mail))
```

Reply:
Thanks for pointing this out!
I adapted these format strings to the surrounding file which uses `%s` for string formatting. I'm happy to change this but would then change it for all instances in this file. I think it's best to do this once the PR has been reviewed by the maintainer to keep the diff noise down. I will keep this comment open as a reminder

Reply:
Agree :) I just wanted to mention it since you asked for feedback on your python code :)

Reply:
Yeah, really grateful for that! As you can see by my gratious use of `==` and `()`, I mostly write JS :laughing: